oci/layout/oci_transport.go

```diff
@@ -85,41 +84,44 @@ type ociReference struct {
 	// (But in general, we make no attempt to be completely safe against concurrent hostile filesystem modifications.)
 	dir         string // As specified by the user. May be relative, contain symlinks, etc.
 	resolvedDir string // Absolute path with no symlinks, at least at the time of its creation. Primarily used for policy namespaces.
-	tag         string
+	image       string
 }
 
 // ParseReference converts a string, which should not start with the ImageTransport.Name prefix, into an OCI ImageReference.
 func ParseReference(reference string) (types.ImageReference, error) {
-	var dir, tag string
-	sep := strings.LastIndex(reference, ":")
-	if sep == -1 {
-		dir = reference
-		tag = "latest"
-	} else {
-		dir = reference[:sep]
-		tag = reference[sep+1:]
+	var dir, image string
+	sep := strings.SplitN(reference, ":", 2)
+	dir = sep[0]
+
+	if len(sep) == 2 {
+		if strings.Contains(sep[1], ":") {
+			image = sep[1]
+		} else {
+			image = sep[1] + ":latest"
+		}
```

Comment:
Sorry, you’re right this is unclear. Yes, that’s what I mean by 1: treat the image name as a string which as no further structure, unlike `docker/reference` where, in the extreme, `busybox` is a shorthand for `docker.io/library/busybox:latest`.

We’ve been treating the ref.name annotation as a Docker-like “tag”, and those default to `latest`; but now there’s the actual spec-by-regexp, which says nothing about existence or default values of tags.

(Not adding the `:latest` also keeps the behavior more, though not completely, compatible with the behavior before this PR.)

[Or is there pre-existing behavior, perhaps in umoci, we should co-opt here? I haven’t checked.]